stubs/seaborn/seaborn/_core/groupby.pyi

```diff
@@ -1,10 +1,30 @@
 from _typeshed import Incomplete
 from collections.abc import Callable
+from typing import Any
+from typing_extensions import Concatenate, ParamSpec
 
 from pandas import DataFrame
+from pandas._typing import (
+    AggFuncTypeFrame,  # pyright: ignore[reportUnknownVariableType]  # This is for pandas-stubs to improve https://github.com/pandas-dev/pandas-stubs/issues/811
+)
+
+_P = ParamSpec("_P")
 
 class GroupBy:
     order: dict[str, list[Incomplete] | None]
     def __init__(self, order: list[str] | dict[str, list[Incomplete] | None]) -> None: ...
-    def agg(self, data: DataFrame, *args, **kwargs) -> DataFrame: ...
-    def apply(self, data: DataFrame, func: Callable[..., DataFrame], *args, **kwargs) -> DataFrame: ...
+    # Signature based on pandas.core.groupby.generic.DataFrameGroupBy.aggregate
+    # args and kwargs possible values depend on func which itself can be
+    # an attribute name, a mapping, a callable, or lead to a jitted numba function
+    def agg(
+        self,
+        data: DataFrame,
+        func: AggFuncTypeFrame = ...,  # pyright: ignore[reportUnknownParameterType]  # This is for pandas-stubs to improve https://github.com/pandas-dev/pandas-stubs/issues/811
+        *args: Any,
```

Comment:
Uncertainty about variance: Would it be fine to use `object` here for `*args` and `**kwargs`?

Reply:
`_typeshed.Incomplete` would probably be the most correct, as we could probably copy over the arguments from `DataFrame.agg()`. (Although I'm not terribly familiar with pandas, so I might be missing something here.)